nautobot/utilities/forms/fields.py

```diff
@@ -676,6 +676,25 @@ def has_changed(self, initial, data):
         return super().has_changed(initial, data)
 
 
+class MultiValueCharField(forms.CharField):
+    """
+    CharField that takes multiple user character inputs and render them as tags in the form field.
+    Press enter to complete an input.
+    """
+
+    widget = widgets.MultiValueCharInput()
+
+    def get_bound_field(self, form, field_name):
+        bound_field = BoundField(form, self, field_name)
+        value = bound_field.value()
+        widget = bound_field.field.widget
+        # Save the selected choices in the widget even after the filterform is submitted
+        if value is not None:
+            widget.choices = [(v, v) for v in value]
+
+        return bound_field
+
+
```

Comment:
Looks like this chain needs to be updated:

https://github.com/nautobot/nautobot/blob/develop/nautobot/utilities/filters.py#L527

https://github.com/nautobot/nautobot/blob/b027b6414c0be1222177793a8691793365354afa/nautobot/utilities/filters.py#L73-L74

https://github.com/nautobot/nautobot/blob/b027b6414c0be1222177793a8691793365354afa/nautobot/utilities/filters.py#L36-L60

Which should fix the magic generation issue, eliminating the need to specify the field.

And is the choices only helping on page reload pre-populate the inputted text? Is the necessary?

Reply:
Yeah the `widget.choices = [(v, v) for v in value] `are there after the filter is applied to keep the user input.